regression/test.pl

```diff
@@ -117,10 +113,27 @@ ($$$$$)
           $included--;
         } else {
           my $r;
-          $result =~ s/\\/\\\\/g;
-          $result =~ s/([^\\])\$/$1\\r\\\\?\$/;
-          system("bash", "-c", "grep $grep_options \$'$result' '$name/$output' >/dev/null");
-          $r = ($included ? $? != 0 : $? == 0);
+
+          # Multi line therefore we run each check across the whole output
+          if($grep_options eq "activate-multi-line-match") {
+            my $dir = getcwd();
+            my $abs_path = "$dir/$name/$output";
+            open( my $fh => $abs_path) || die "Cannot open '$name/$output': $!";
+
+            local $/ = undef;
+            binmode $fh;
+            my $whole_file = <$fh>;
+            my $is_match = $whole_file =~ /$result/;
+            $r = ($included ? !$is_match : $is_match);
+            close($fh);
+          }
+          else
+          {
+            $result =~ s/\\/\\\\/g;
+            $result =~ s/([^\\])\$/$1\\r\\\\?\$/;
+            system("bash", "-c", "grep \$'$result' '$name/$output' >/dev/null");
```

Comment:
That was a bit of a rabbit hole, for clarity I've checked it in as a revert + a separate commit fixing the tests. 

The bulk changes I made were to:

- escape/unescape `(`, `)` style brackets
- escape/unescape `+`, `|`

Then there were some more fiddly ones in the tests that actually run: 
- don't escape regex characters when in `std=c++11` in the flags section
- Escape $ if in the middle of the line (e.g. in Quantifiers-not-exists which has a dollar in the middle and the end of the line and grep is just matching up to the first one and accepting it):

As such, I make no guarantees that tests in the other folders will behave the same way...

Once you've reviewed it, I'll drop out the original commit and the revert. Would you like me to squash the test changes into the first commit (so the commit passes) or leave it as separate so easy to see the changes made to the tests)?